server/src/test/java/org/elasticsearch/indices/IndicesModuleTests.java

```diff
@@ -87,26 +90,50 @@ public MetadataFieldMapper getDefault(MappedFieldType fieldType, ParserContext c
             RoutingFieldMapper.NAME, IndexFieldMapper.NAME, SourceFieldMapper.NAME, TypeFieldMapper.NAME,
             VersionFieldMapper.NAME, SeqNoFieldMapper.NAME, FieldNamesFieldMapper.NAME};
 
+    private static String[] EXPECTED_METADATA_FIELDS_6x = new String[]{AllFieldMapper.NAME, IgnoredFieldMapper.NAME,
+        IdFieldMapper.NAME, RoutingFieldMapper.NAME, IndexFieldMapper.NAME, SourceFieldMapper.NAME, TypeFieldMapper.NAME,
+        VersionFieldMapper.NAME, SeqNoFieldMapper.NAME, FieldNamesFieldMapper.NAME};
+
+
     public void testBuiltinMappers() {
         IndicesModule module = new IndicesModule(Collections.emptyList());
-        assertFalse(module.getMapperRegistry().getMapperParsers().isEmpty());
-        assertFalse(module.getMapperRegistry().getMetadataMapperParsers().isEmpty());
-        Map<String, MetadataFieldMapper.TypeParser> metadataMapperParsers = module.getMapperRegistry().getMetadataMapperParsers();
-        int i = 0;
-        for (String field : metadataMapperParsers.keySet()) {
-            assertEquals(EXPECTED_METADATA_FIELDS[i++], field);
+        {
+            Version version = VersionUtils.randomVersionBetween(random(), Version.V_6_0_0, Version.V_7_0_0.minimumCompatibilityVersion());
+            assertFalse(module.getMapperRegistry().getMapperParsers().isEmpty());
+            assertFalse(module.getMapperRegistry().getMetadataMapperParsers(version).isEmpty());
+            Map<String, MetadataFieldMapper.TypeParser> metadataMapperParsers =
+                module.getMapperRegistry().getMetadataMapperParsers(version);
+            int i = 0;
+            for (String field : metadataMapperParsers.keySet()) {
+                assertEquals(EXPECTED_METADATA_FIELDS_6x[i++], field);
+            }
```

Comment:
do we need to check that the have the same size as well?

Reply:
https://github.com/elastic/elasticsearch/pull/37808/commits/cdaa804e87ec5a7a276d31e6a71eb06d8d69924c